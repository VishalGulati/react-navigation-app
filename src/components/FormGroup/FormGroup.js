import React from 'react';
import PropTypes from 'prop-types';
import AutoCompleteInput from '../../containers/AutoCompleteInput/AutoCompleteInput';

/**
 * A re-usable comonent that returns a combination of label and an input field 
 * collectively called form group
 */
const FormGroup = props => {
  return (
    <div className="form-group">
      <label htmlFor={props.inputId}>{props.label}</label>
      <AutoCompleteInput inputId={props.inputId} />
    </div>
  );
};

FormGroup.propTypes = {
  inputId: PropTypes.string,
  label: PropTypes.string
};

export default FormGroup;
