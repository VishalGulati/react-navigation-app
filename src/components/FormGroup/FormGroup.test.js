import React from 'react';
import ReactDOM from 'react-dom';
import FormGroup from './FormGroup';
import { shallow } from 'enzyme';

const wrapper = shallow(<FormGroup />);

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<FormGroup />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('Component has container div with class form-group', () => {
    expect(wrapper.find('div').hasClass('form-group')).toEqual(true);
});


it('snapshot testing', () => {
    expect(wrapper).toMatchSnapshot();
});