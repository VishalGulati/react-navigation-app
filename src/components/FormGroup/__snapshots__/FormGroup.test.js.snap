// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormGroup />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label />,
        <AutoCompleteInput
          className="form-control"
        />,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "htmlFor": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "form-control",
          "inputId": undefined,
          "mapLoaded": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label />,
          <AutoCompleteInput
            className="form-control"
          />,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "htmlFor": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "form-control",
            "inputId": undefined,
            "mapLoaded": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
